ci/docker-hub-auth.sh

```
@@ -11,20 +11,12 @@
 # testcontainers' RegistryAuthLocator does `if (getenv != null) ObjectMapper.readTree(...)`, so an
 # empty string would throw a JSON parse error; only a genuinely absent var is skipped cleanly.
 #
-# Inputs are read from the env (the caller maps the Azure pipeline vars):
+# Inputs are read from the env:
 #   DOCKER_HUB_USER, DOCKER_HUB_RO_TOKEN
-#
-# An undefined Azure non-secret var expands to its literal "$(NAME)"; reject that form too.
-if [ -n "${DOCKER_HUB_RO_TOKEN:-}" ] && [ "${DOCKER_HUB_RO_TOKEN}" != '$(DOCKER_HUB_RO_TOKEN)' ] \
-   && [ -n "${DOCKER_HUB_USER:-}" ] && [ "${DOCKER_HUB_USER}" != '$(DOCKER_HUB_USER)' ]; then
+if [ -n "${DOCKER_HUB_RO_TOKEN:-}" ] && [ -n "${DOCKER_HUB_USER:-}" ]; then
   export DOCKER_AUTH_CONFIG="{\"auths\":{\"https://index.docker.io/v1/\":{\"auth\":\"$(printf '%s:%s' "$DOCKER_HUB_USER" "$DOCKER_HUB_RO_TOKEN" | base64 -w0)\"}}}"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)